flask/app/families.py

```diff
@@ -16,27 +16,50 @@
 @family_blueprint.route("/api/families", methods=["GET"])
 @login_required
 def list_families():
+
+    app.logger.debug("Parsing query parameters..")
+
     starts_with = request.args.get("starts_with", default="", type=str)
     starts_with = f"{starts_with}%"
     max_rows = request.args.get("max_rows", default=100)
     order_by_col = request.args.get("order", default="family_id", type=str)
+
+    app.logger.debug(
+        "Query parameters: \n\tstarts_with %s \n\tmax_rows %s, \n\torder_by_col %s",
+        starts_with,
+        max_rows,
+        order_by_col,
+    )
+
+    app.logger.debug("Validating max rows..")
     try:
         int(max_rows)
     except:
+        app.logger.error("Max Rows: %s is not a valid integer")
         abort(400, description="Max rows must be a valid integer")
 
     columns = models.Family.__table__.columns.keys()
 
+    app.logger.debug("Validating 'order_by' column..")
     if order_by_col not in columns:
+        app.logger.error(
+            "Column name %s is not valid, it must be one of %s", order_by_col, columns
+        )
         abort(400, description=f"Column name for ordering must be one of {columns}")
     column = getattr(models.Family, order_by_col)
 
+    app.logger.debug("Query parameters successfully parsed.")
+
+    app.logger.debug("Retrieving user id..")
     if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
         user_id = request.args.get("user")
+        app.logger.debug("User is admin with ID '%s'", user_id)
+        app.logger.warning("LOGIN_DISABLED = TRUE may also be set.")
```

Comment:
We shouldn't warn here or we'll warn every time an admin performs an action. `user_id` is the assumed identity if any, otherwise `None` causes admin behaviour later.